public/install/functions.php

```diff
@@ -17,126 +25,196 @@ function checkPhpVersion()
         || $_SERVER['SERVER_PORT'] == 443;
 }
 
-function getMySQLVersion()
+/**
+ * Check if MySQL is installed and runs the correct version using a shell command
+ * @return mixed|string 'OK' if required version is met, returns MySQL version if not met.
+ */
+function getMySQLVersion(): mixed
 {
     global $requirements;
 
-    $output = shell_exec('mysql -V');
+    $output = shell_exec('mysql -V') ?? '';
     preg_match('@[0-9]+\.[0-9]+\.[0-9]+@', $output, $version);
 
     $versionoutput = $version[0] ?? '0';
 
     return intval($versionoutput) > intval($requirements['mysql']) ? 'OK' : $versionoutput;
 }
 
-function getZipVersion()
+/**
+ * Check if zip is installed using a shell command
+ * @return string 'OK' on success and 'not OK' on failure.
+ */
+function getZipVersion(): string
 {
-    $output = shell_exec('zip  -v');
+    $output = shell_exec('zip  -v') ?? '';
     preg_match('@[0-9]+\.[0-9]+\.[0-9]+@', $output, $version);
 
     $versionoutput = $version[0] ?? 0;
 
     return $versionoutput != 0 ? 'OK' : 'not OK';
 }
 
-function getGitVersion()
+/**
+ * Check if git is installed using a shell command
+ * @return string 'OK' on success and 'not OK' on failure.
+ */
+function getGitVersion(): string
 {
-    $output = shell_exec('git  --version');
+    $output = shell_exec('git  --version') ?? '';
     preg_match('@[0-9]+\.[0-9]+\.[0-9]+@', $output, $version);
 
     $versionoutput = $version[0] ?? 0;
 
     return $versionoutput != 0 ? 'OK' : 'not OK';
 }
 
-function getTarVersion()
+/**
+ * Check if tar is installed using a shell command
+ * @return string 'OK' on success and 'not OK' on failure.
+ */
+function getTarVersion(): string
 {
-    $output = shell_exec('tar  --version');
+    $output = shell_exec('tar  --version') ?? '';
     preg_match('@[0-9]+\.[0-9]+@', $output, $version);
 
     $versionoutput = $version[0] ?? 0;
 
     return $versionoutput != 0 ? 'OK' : 'not OK';
 }
 
-function checkExtensions()
+/**
+ * Check all extensions to see if they have loaded or not
+ * @return array Returns an array of extensions that failed to load.
+ */
+function checkExtensions(): array
 {
-    global $required_extentions;
+    global $required_extensions;
 
     $not_ok = [];
     $extentions = get_loaded_extensions();
 
-    foreach ($required_extentions as $ext) {
-        if (! preg_grep('/^(?=.*'.$ext.').*$/', $extentions)) {
+    foreach ($required_extensions as $ext) {
+        if (!preg_grep('/^(?=.*' . $ext . ').*$/', $extentions)) {
             array_push($not_ok, $ext);
         }
     }
 
     return $not_ok;
 }
 
-function setEnvironmentValue($envKey, $envValue)
+/**
+ * Sets the environment variable into the env file
+ * @param string $envKey The environment key to set or modify
+ * @param string $envValue The environment variable to set
+ * @return bool true on success or false on failure.
+ */
+function setEnvironmentValue(string $envKey, $envValue)
```

Comment:
Just using putenv/getenv and put it into a new function seems utterly redundant to me.

Reply:
That was there because i had not refactored the actual functions in each file yet

Reply:
I've refactored getenv, removed the function. Refactored setenv however, due to the way the code looks using "=" is not ideal. I prefer to keep the function method for setting environmental values.